str/runners/str_iterative_eh_runner.py

```diff
@@ -160,6 +160,9 @@ def main(
             )
             # per sample, run parallel jobs on each shard of the catalog
             for index, subcatalog in enumerate(catalog_files, start=1):
+
+                if to_path(output_path(f'{cpg_id}/{cpg_id}_eh_shard{index}', 'analysis')).exists():
```

Comment:
You should save this string as a variable and reuse below

Reply:
It's a master path; I should check for .vcf specifically.